### PR TITLE
Feature #111 dwaa support

### PR DESCRIPTION
**Pull request by kalisp**
Exr compressed with DWAA are decompressed first and only after consumed by FFMPEG.

Currently its using flag `multipartExr` on instance to recognize compressed files, which might not be enough/might be wrong.

Currently implemented only `extract_review`, `extract_burnin`, `extract_jpeg` are in the process.

(Still works only on Windows, for unsupported/without oiio tool platforms original process should be still used.)

### CODE REVIEW

**Comment by tokejepsen** [comment]
I would suggest enforcing decompression to local temp directory for clean up later and speed of reading.

**Comment by mkolar** [comment]
> I would suggest enforcing decompression to local temp directory for clean up later and speed of reading.

I agree

**Comment by tokejepsen** [comment]
I would also suggest only decompressing when its incompatible with FFMPEG, which you can check with:

```
        output = lib._subprocess(["oiiotool", "--info", "-v", first_file])
        if "compression: \"dwaa\"" in output:
```

Lastly I would only check the first file in a sequence and assume all frames in that sequence is encoded the same. Then you can run the decompression on all the frames at once with OIIO's `-frames` flag.

**Comment by kalisp** [comment]
Implemented all Toke's notes (thanks for them).

Not implemented yet:
`extract_burnin`
`extract_review.legacy`

File: pype/plugins/global/publish/extract_jpeg.py [hunk 1]
```diff
@@ -60,6 +60,19 @@ def process(self, instance):
             full_input_path = os.path.join(stagingdir, input_file)
             self.log.info("input {}".format(full_input_path))
 
+            decompressed_dir = ''
+            decompress = plugins_lib.should_decompress(full_input_path)
+            if decompress:
+                decompressed_dir = plugins_lib.get_decompress_dir()
```
Comment (jezscha) [inline pype/plugins/global/publish/extract_jpeg.py:66]: Is there some particular reason to have function for returning `decompressed_dir` outside of `decompress` function? Wouldn't it be easier to use it inside `decompress` function and return the `decompressed_dir` from `decompress` function?

Comment (kalisp) [reply pype/plugins/global/publish/extract_jpeg.py:66]: For me it looks cleaner, function should do one thing and one thing only. This way you have more control, its more testable, no hidden side effects..imho

**Comment by kalisp** [comment]
Implemented `extract_burnin`.

Currently only legacy parts not implemented, not sure if they are still used and applicable for this.

**Comment by kalisp** [comment]
Fixed support for Python 2 hosts (Maya).

**Pull request merged**